### PR TITLE
feat: add directDepsLeadingTo public method

### PR DESCRIPTION
**Pull request by AndreDalcher**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/dep-graph/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Add a public method to get the direct dependencies which introduced the specified package.

### CODE REVIEW

**Comment by darscan** [comment]
Can we get a deeper chain in the test fixture? (where there is at least one intermediate pkg in the chain between the direct and the target)

**Comment by michael-go** [comment]
please update the README too as it lists the DepGraph methods

File: src/core/dep-graph.ts [hunk 1]
```diff
@@ -183,6 +183,16 @@ class DepGraphImpl implements types.DepGraphInternal {
     return this.nodeEquals(this, this.rootNodeId, otherDepGraph, otherDepGraph.rootNodeId, compareRoot);
   }
 
+  public directDepsLeadingTo(pkg: types.Pkg): types.PkgInfo[] {
+    const pkgNodes = this.getPkgNodeIds(pkg);
+    const directDeps = this.getNodeDepsNodeIds(this.rootNodeId);
+    const nodes = directDeps.filter((directDep) => {
+      const reachableNodes = graphlib.alg.postorder(this._graph, [directDep]);
```
Comment (michael-go) [inline src/core/dep-graph.ts]: please check (and add a test) if `graphlib.alg.postorder` works with cyclic graphs. If not, please check for cycles and throw if exist like done in some other places in the lib.
This also means this method will need to be re-implemented differently as part of https://github.com/snyk/dep-graph/pull/48 (but it's OK)

Comment (AndreDalcher) [reply src/core/dep-graph.ts]: It seems to work :D

**Pull request merged**